src/Symfony/Component/Messenger/Tests/Transport/RedisExt/RedisExtIntegrationTest.php

```diff
@@ -104,4 +102,77 @@ public function testConnectionSendDelayedMessagesWithSameContent()
         $this->assertEquals($body, $encoded['body']);
         $this->assertEquals($headers, $encoded['headers']);
     }
+
+    public function testConnectionBelowRedeliverTimeout()
+    {
+        // lower redeliver timeout and claim interval
+        $connection = Connection::fromDsn(getenv('MESSENGER_REDIS_DSN'), [], $this->redis);
+
+        $connection->cleanup();
+        $connection->setup();
+
+        $body = '{"message": "Hi"}';
+        $headers = ['type' => DummyMessage::class];
+
+        // Add two messages
+        $connection->add($body, $headers);
+
+        // Read first message with other consumer
+        $this->redis->xreadgroup(
+            'symfony',
+            'other-consumer2',
+            ['messages' => '>'],
+            1
+        );
+
+        // Queue will not have any messages yet
+        $this->assertNull($connection->get());
+
+        sleep(10);
+
+        // Queue will not have any messages yet
+        $this->assertNull($connection->get());
+    }
+
+    public function testConnectionClaimAndRedeliver()
+    {
+        // lower redeliver timeout and claim interval
+        $connection = Connection::fromDsn(
+            getenv('MESSENGER_REDIS_DSN'),
+            ['redeliver_timeout' => 0, 'claim_interval' => 500],
+            $this->redis
+        );
+
+        $connection->cleanup();
+        $connection->setup();
+
+        $body1 = '{"message": "Hi"}';
+        $body2 = '{"message": "Bye"}';
+        $headers = ['type' => DummyMessage::class];
+
+        // Add two messages
+        $connection->add($body1, $headers);
+        $connection->add($body2, $headers);
+
+        // Read first message with other consumer
+        $this->redis->xreadgroup(
+            'symfony',
+            'other-consumer2',
+            ['messages' => '>'],
+            1
+        );
+
+        // Queue will return the pending message first because redeliver_timeout = 0
+        $encoded = $connection->get();
```

Comment:
this test is flaky in my case sometimes I get `Hi` and sometimes I get `Bye` as first message. Not sure why 🙈